app/models/hathitrust.rb

```diff
@@ -129,45 +129,55 @@ def check_async(task)
     ecs.run_task(args)
   end
 
-  private
-
+  
   ##
   # Downloads the latest HathiFile.
   #
   # @return The path of the HathiFile
   #
+      
+  private
+
   def get_hathifile(task)
     # As there is no single URI for the latest HathiFile, we have to scrape
     # the HathiFile listing out of the index HTML page.
     task.update!(name: 'Checking HathiTrust: downloading HathiFile index...')
     puts task.name
 
-    uri      = URI.parse('https://www.hathitrust.org/hathifiles')
-    response = Net::HTTP.get_response(uri)
-    page     = Nokogiri::HTML(response.body)
+    base_url     = 'https://www.hathitrust.org'
+    response     = Net::HTTP.get_response(URI('https://www.hathitrust.org/hathifiles'))
+
+    location     = response['location']
+    res          = base_url + location 
+
+    new_response = Net::HTTP.get_response(URI(res))
+
```

Comment:
This will work as long as `https://www.hathitrust.org/hathifiles` continues redirecting. But if the HathiFiles page ever moves back to that URL, there will be no redirect, which will break the code. You should check the response status code and only send another request if it's 3xx (300-399).

Reply:
I added a conditional block so another request is sent only if the response status code is a `3xx`. When I run the check locally it succeeds with this additional code block (screenshot attached). 

I also went through and fixed the issues with indentation/white space etc. Thank you for pointing that out - I will be more mindful to check for these things moving forward. Hopefully I caught them all, but please let me know if you still notice any issues. 
<img width="934" alt="Screen Shot 2023-09-22 at 10 13 58 AM" src="https://github.com/medusa-project/book-tracker/assets/103534307/0f2dfc85-c44b-4104-8d5f-ba6aba897348">